src/main/java/com/google/idea/perf/methodtracer/TracerConfig.kt

```diff
@@ -44,103 +53,179 @@ object TracerConfig {
-                    classJvmName, methodName, methodDesc, TracepointProperties.DEFAULT
-                )
-                tracepoints.append(tracepoint)
-            }
 
-            val tracepoint = getTracepoint(methodId)
-            tracepoint.parameters.set(0)
-            tracepoint.setFlags(TracepointFlags.TRACE_ALL)
-        }
-    }
+            // If the class config isn't loaded, enqueue a command.
+            classConfig.commands?.add(Pair(methodRegex, TracepointProperties(flags, parameterBits)))
 
-    fun untraceMethods(classJvmName: String, methodName: String) {
-        lock.withLock {
-            val classConfig = classConfigs[classJvmName] ?: return
+            // If the class config is loaded, set tracepoint properties.
+            for ((signature, methodId) in classConfig.methodIds) {
+                if (methodId != null) {
+                    val methodName = signature.substringBefore('(')
 
-            for ((signature, _) in classConfig.methodIds) {
-                if (signature.substringBefore('(') == methodName) {
-                    val methodDesc = signature.substring(methodName.length)
-                    val methodId = getMethodId(classJvmName, methodName, methodDesc)
-                    if (methodId != null) {
+                    if (methodRegex.matcher(methodName).matches()) {
                         val tracepoint = getTracepoint(methodId)
-                        tracepoint.unsetFlags(TracepointFlags.TRACE_ALL)
-                        tracepoint.parameters.set(0)
+                        tracepoint.parameters.set(parameterBits)
+                        if (enable) {
+                            tracepoint.setFlags(flags)
+                        }
+                        else {
+                            tracepoint.unsetFlags(TracepointFlags.TRACE_ALL)
+                        }
                     }
                 }
             }
+
+            return listOf(className)
         }
     }
 
-    fun untraceMethod(method: Method) {
-        lock.withLock {
-            val classJvmName = method.declaringClass.name.replace('.', '/')
-            val methodDesc = Type.getMethodDescriptor(method)
-            untraceMethod(classJvmName, method.name, methodDesc)
+    private fun setTrace(
+        enable: Boolean,
+        classPattern: String,
+        flags: Int = 0,
+        parameters: Collection<Int> = emptyList()
+    ): List<String> {
+        val regex = Pattern.compile(PatternUtil.convertToRegex(classPattern))
+        val matcher = regex.matcher("")
+
+        if (enable) {
+            val classes = AgentLoader.instrumentation?.allLoadedClasses ?: return emptyList()
+
+            val matchingClasses = mutableListOf<String>()
+            for (clazz in classes) {
+                val className = clazz.name
+                matcher.reset(className)
+                if (matcher.matches()) {
+                    matchingClasses.add(className)
+                }
+            }
```

Comment:
Since there are >37,000 loaded classes, the filtering function has shown poor performance. Using a manual for loop seems to fix this issue.